wandb/sdk/launch/sweeps/scheduler.py

```diff
@@ -0,0 +1,206 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+import logging
+import os
+from typing import Any, Dict, List, Optional
+
+import wandb
+from wandb.apis.internal import Api
+import wandb.apis.public as public
+from wandb.sdk.launch.launch_add import launch_add
+from wandb.sdk.lib.runid import generate_id
+
+logger = logging.getLogger(__name__)
+
+
+class SchedulerState(Enum):
```

Comment:
is this needed or can this be reflected via the launch run states? since the scheduler itself is just a run

Reply:
Yeah there is definitely a bunch of overlap between states for runs, jobs, scheduler, etc. I figured having a separate state for the Scheduler would be useful in the future when there are a bunch of types of schedulers. But we can always change it to match the Job state in the future.